source/postprocess/visualization/seismic_melt.cc

```diff
@@ -0,0 +1,372 @@
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/postprocess/visualization/seismic_melt.h>
+#include <aspect/melt.h>
+#include <aspect/utilities.h>
+#include <aspect/simulator.h>
+#include <aspect/material_model/interface.h>
+
+#include <deal.II/numerics/data_out.h>
+
+
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace VisualizationPostprocessors
+    {
+
+      template <int dim>
+      MeltSeismicProperties<dim>::
+      MeltSeismicProperties ()
+        :
+        DataPostprocessor<dim> ()
+      {}
+
+      template <int dim>
+      std::vector<std::string>
+      MeltSeismicProperties<dim>::
+      get_names () const
+      {
+        std::vector<std::string> solution_names;
+
+        for (unsigned int i=0; i<property_names.size(); ++i)
+          {
+            solution_names.push_back(property_names[i]);
+            std::replace(solution_names.back().begin(),solution_names.back().end(),' ', '_');
+          }
+        return solution_names;
+      }
+
+      template <int dim>
+      std::vector<DataComponentInterpretation::DataComponentInterpretation>
+      MeltSeismicProperties<dim>::
+      get_data_component_interpretation () const
+      {
+        std::vector<DataComponentInterpretation::DataComponentInterpretation> interpretation;
+        for (unsigned int i=0; i<property_names.size(); ++i)
+          {
+            interpretation.push_back (DataComponentInterpretation::component_is_scalar);
+          }
+        return interpretation;
+      }
+
+      template <int dim>
+      UpdateFlags
+      MeltSeismicProperties<dim>::
+      get_needed_update_flags () const
+      {
+        return  update_values  | update_q_points;
+      }
+
+      template <int dim>
+      void
+      MeltSeismicProperties<dim>::
+      evaluate_vector_field(const DataPostprocessorInputs::Vector<dim> &input_data,
+                            std::vector<Vector<double> > &computed_quantities) const
+      {
+        AssertThrow(this->include_melt_transport()==true,
+                    ExcMessage("'Include melt transport' has to be on when using melt transport postprocessors."));
+
+        const unsigned int n_quadrature_points = input_data.solution_values.size();
+        Assert (computed_quantities.size() == n_quadrature_points,    ExcInternalError());
+        Assert (input_data.solution_values[0].size() == this->introspection().n_components,   ExcInternalError());
+
+        // Set use_strain_rates to true since the compaction viscosity might also depend on the strain rate.
+        MaterialModel::MaterialModelInputs<dim> in(input_data,
+                                                   this->introspection(), true);
```

Comment:
I think you can set this to `false` instead of `true`, as you only use the densities (not the viscosities) from the material model outputs, and change the comment above to something like:
'We don't need the strain rate, because we only evaluate the solid and melt densities'